### PR TITLE
Google Scholar Cite

### LINKED ISSUE #174: Cite Results

### PR DESCRIPTION
**Pull request by Mykhailo-Zub**
Resolves #174. [Draft](https://serpapi.com/blog/p/2645ec01-c628-4f52-97e0-653af5b660f0/)

### CODE REVIEW

File: Google/google_scholar/organic_cite_results/javascript/cite_results/google-scholar-cite-results-scrape.md [hunk 1]
```diff
@@ -0,0 +1,517 @@
...
+
+const domain = `http://scholar.google.com`;
+const pagesLimit = Infinity;
+let currentPage = 1;
+```
+
+|Code|Explanation|
+|----|-----------|
+|`q`|search query|
+|`hl`|parameter defines the language to use for the Google search|
```
Comment (dimitryzub) [inline Google/google_scholar/organic_cite_results/javascript/cite_results/google-scholar-cite-results-scrape.md:148]:
```suggestion
|`hl`|parameter defines the language to use for the Google Scholar search|
```

File: Google/google_scholar/organic_cite_results/javascript/cite_results/google-scholar-cite-results-scrape.md [hunk 2]
```diff
@@ -0,0 +1,517 @@
...
+|`page.setDefaultNavigationTimeout(60000)`|changing default (30 sec) time for waiting for selectors to 60000 ms (1 min) for slow internet connection|
+|`page.goto(URL)`|navigation to `URL` which is defined above|
+|`browser.close()`|after all we close the browser instance|
+|`console.dir(result, { depth: null })`|console method `dir` allows you to use an object with the necessary parameters to change default output options. Watch [Node.js documentation](https://nodejs.org/api/console.html#consoledirobj-options) for more info|
+
+Now we can launch our parser. To do this enter `node YOUR_FILE_NAME` in your command line. Where `YOUR_FILE_NAME` is the name of your `.js` file.
+
+<h2 id='output'>Output</h2>
+
+📌Note: if you see something like `[Object]` in your console you can use `console.dir(result, { depth: null })` instead `console.log()`. Watch [Node.js documentation](https://nodejs.org/api/console.html#consoledirobj-options) for more info.
```
Comment (dimitryzub) [inline Google/google_scholar/organic_cite_results/javascript/cite_results/google-scholar-cite-results-scrape.md:267]: 👍👍

File: Google/google_scholar/organic_cite_results/javascript/cite_results/google-scholar-cite-results-scrape.md [hunk 3]
```diff
@@ -0,0 +1,517 @@
...
+      ]
+   },
+   ...and other results
+]
+```
+
+<h2 id='links'>Links</h2>
+
+* [Code in the online IDE](https://replit.com/@MikhailZub/Google-Scholar-Cite-NodeJS-SerpApi#index.js) 
+* [Google Scholar API](https://serpapi.com/google-scholar-api)
```
Comment (dimitryzub) [inline Google/google_scholar/organic_cite_results/javascript/cite_results/google-scholar-cite-results-scrape.md:511]:
```suggestion
* [Google Scholar Cite API](https://serpapi.com/google-scholar-cite-api)
```

**Review by dimitryzub — CHANGES_REQUESTED** [review]
@Mykhailo-Zub Thank you for the quality step-by-step tutorial blog post 💯

On replit, when individual cite results are being parsed, after extracting cite ID I get error but it seems to be a replit thing:
image

There're also a few minor things I've noticed.

**Comment by Mykhailo-Zub** [comment]
> On replit, when individual cite results are being parsed, after extracting cite ID I get error but it seems to be a replit thing:

@dimitryzub I wrapped this code in `try-catch` to prevent crashing the script if we get some error.

**Review by dimitryzub — APPROVED** [review]
@Mykhailo-Zub Looks good 💯Thank you!

You can publish this blog post whenever you can and merge this PR. Would it be possible to publish it this week (July 11-15)?

**Comment by Mykhailo-Zub** [comment]
https://serpapi.com/blog/how-to-scrape-google-scholar-cite-with-nodejs/
https://medium.com/@mikhail.a.zub/web-scraping-google-scholar-cite-results-with-nodejs-64abef8da369
https://dev.to/serpapi/web-scraping-google-scholar-cite-results-with-nodejs-3ncg
https://serpapi.hashnode.dev/web-scraping-google-scholar-cite-results-with-nodejs

**Pull request merged**